saleor/graphql/order/mutations/fulfillments.py

```diff
@@ -119,18 +141,57 @@ def clean_lines(cls, order_lines, quantities):
                     }
                 )
 
+    @classmethod
+    def check_warehouses_for_duplicates(cls, warehouse_ids):
+        for warehouse_ids_for_line in warehouse_ids:
+            duplicates = {
+                id
+                for id in warehouse_ids_for_line
+                if warehouse_ids_for_line.count(id) > 1
+            }
```

Comment:
The same here, you can use `get_duplicated_values`:
https://github.com/mirumee/saleor/blob/master/saleor/graphql/core/utils/__init__.py#L115